workflows/speciesabundance.nf

```diff
@@ -68,19 +68,39 @@ workflow SpAnce {
 
     ch_versions = Channel.empty()
 
+    // Track processed IDs
+    def processedIDs = [] as Set
+
     // Create a new channel of metadata from a sample sheet
     // NB: `input` corresponds to `params.input` and associated sample sheet schema
     input = Channel.fromSamplesheet("input")
+        .map { meta, fastq_1, fastq_2 ->
+            // Set meta.id to meta.irida_if if sample_name is not provided in the samplesheet
+            if (!meta.id) {
+                meta.id = meta.irida_id
+            } else {
+                // Replace spaces in 'id' with underscores
+                meta.id = meta.id.replaceAll(/\s+/, '_')
```

Comment:
Instead of just spaces, could you replace any non-alphanumeric (and a few special characters) with underscores? E.g., something like `/[^A-Za-z0-9_-./`?

That way, we guarantee that only characters we know won't mess up commands will be used as the identifiers.

Reply:
Would this introduce the potential to have 2 identical IDs after replacement? Should we test for this?

- "ID@" -> "ID_"
- "ID#" -> "ID_"

Reply:
Alphanumeric syntax change made here: [cc6a23d](https://github.com/phac-nml/speciesabundance/pull/24/commits/cc6a23d7fba904b92ee01f29bddfee0aec760666)

Thanks Eric! Great suggestion to ensure replacement of non-alphanumeric characters can still be unique! This variation of the `sample_name` is included in the added test here: [5dd474e](https://github.com/phac-nml/speciesabundance/pull/24/commits/5dd474e717ea09f6f93c08486d5833b1245d6519)